pkg/agent/util/net_linux.go

```diff
@@ -376,3 +378,151 @@ func renameHostInterface(oriName string, newName string) error {
 	}
 	return nil
 }
+
+func interfaceExists(name string) (bool, error) {
```

Comment:
We can just use `netInterfaceByName` and check the return value?

Reply:
I thought about it but the errors are [not exported by net pkg](https://github.com/golang/go/blob/go1.22.0/src/net/interface.go#L24), I would have to do a string match on error string to see if doesn't exists which I guess is not a good way.

Reply:
Ok. In this case, I am fine with your approach.